### PR TITLE
[INLONG-9134][Agent] Add file related utils

### LINKED ISSUE #9134: [Improve][Agent] Add file related utils
### Description

Add file related utils 

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
### Prepare a Pull Request
[INLONG-9134][Agent] Add file related utils

- Fixes #9134 

### Motivation

Add file related utils

### Modifications

Add file related utils

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.agent.plugin.utils.file;
+
+/**
+ * Created by lamberliu on 2015/11/13.
```
Comment (doleyzi) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java:21]: The information here needs to be updated

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java:21]: ok

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/FilePathUtil.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        ret.add(baseDirectory);
+        ret.add(regixDirectory);
+        ret.add(fileName);
+        return ret;
+    }
+
+    public static boolean isSameDir(String fileName1, String fileName2) {
+        ArrayList<String> ret1 = FilePathUtil.cutDirectory(fileName1);
+        ArrayList<String> ret2 = FilePathUtil.cutDirectory(fileName2);
+        if (ret1.get(0).equals(ret2.get(0))) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/FilePathUtil.java:163]: return ret1.get(0).equals(ret2.get(0))

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/Files.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+        }
+        return createTime;
+    }
+
+    public static long getFileLastModifyTime(String fileName) {
+        long lastModify = 0L;
+        try {
+            lastModify = java.nio.file.Files.getLastModifiedTime(Paths.get(fileName)).toMillis();
+        } catch (IOException ioe) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/Files.java:66]: catch without handle

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+package org.apache.inlong.agent.plugin.utils.file;
+
+public class MatchPoint {
+
+    String str;
+    int start;
+    int end;
+
+    MatchPoint(String str1, int start1, int end1) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/MatchPoint.java:26]: please don't use number in parameter name

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/FilePathUtil.java (hunk 2 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/FilePathUtil.java:163]: ok

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 5]
```diff
@@ -0,0 +1,955 @@
...
+    private static final String LIMIT_SEP = "(?<=[a-zA-Z])";
+    private static final String LETTER_STR = "\\D+";
+    private static final String DIGIT_STR = "[0-9]+";
+    private static final Pattern pattern = Pattern.compile(TIME_REGEX,
+            Pattern.CASE_INSENSITIVE | Pattern.DOTALL | Pattern.MULTILINE);
+    private static final Pattern bracePatt = Pattern.compile("\\{(.*?)\\}");
+    private static final int DEFAULT_LENGTH = "yyyyMMddHHmm".length();
+    public static long DAY_TIMEOUT_INTERVAL = 2 * 24 * 3600 * 1000;
+    public static long HOUR_TIMEOUT_INTERVAL = 2 * 3600 * 1000;
+    // 数据源配置异常 */
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:54]: chinese

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 6]
```diff
@@ -0,0 +1,955 @@
...
+        } catch (ParseException e) {
+            return null;
+        }
+
+        /*
+         * The delay should be added to the data time, so remove the - from offset.
+         */
+        if (offset.startsWith("-")) {
+            offset = offset.substring(1, offset.length());
+        } else { // 为正，配置提前读取文件
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:102]: ditto

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 7]
```diff
@@ -0,0 +1,955 @@
...
+            retTime = NewDateUtils.millSecConvertToTimeStr(retCalendar.getTime().getTime(),
+                    cycleUnit);
+        } catch (Exception e) {
+            logger.error("getDateTime error: ", e);
+        }
+        return retTime;
+    }
+
+    public static String getDateTime(long time, String cycleUnit, String offset) {
+        // String retTime = NewDateUtils.millSecConvertToTimeStr(System.currentTimeMillis(),
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:143]: remove useless code

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 8]
```diff
@@ -0,0 +1,955 @@
...
+        // cycleUnit);
+
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(time);
+        Calendar retCalendar = getDateTime(calendar, cycleUnit, offset);
+        return NewDateUtils.millSecConvertToTimeStr(retCalendar.getTime().getTime(), cycleUnit);
+
+        // return retTime;
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:151]: ditto

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 9]
```diff
@@ -0,0 +1,955 @@
...
+
+        String offsetUnit = offset.substring(offset.length() - 1, offset.length());
+        int offsetNumber = Integer.parseInt(offset.substring(0, offset.length() - 1));
+
+        /*
+         * For day task, the offset cycle unit can only be day; for hourly task, the offset can't be minute; for
+         * minutely task, the offset cycle unit can be day, hour and minute, but if the offset cycle unit is minute, the
+         * offset must be divided by cycle number.
+         */
+        // if (("D".equalsIgnoreCase(cycleUnit) && !"D".equalsIgnoreCase(offsetUnit))) {
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:167]: ditto

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/Files.java (hunk 3 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/Files.java:66]: never use, delete it

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 5 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:54]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 6 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:102]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 7 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:143]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java (hunk 9 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:167]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java [hunk 10]
```diff
@@ -0,0 +1,937 @@
...
+            timeInterval += caclOffset(timeOffset);
+        } else { // 处理向后偏移
+            timeInterval -= caclOffset(timeOffset);
+        }
+
+        return isValidCreationTime(dataTime, timeInterval);
+    }
+
+    /**
+     * 根据偏移量计算偏移时间
```
[the remark this answers is not in the document]
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/utils/file/NewDateUtils.java:234]: fix it in next issue

**Review by EMsnap — APPROVED** [review]

**Review by gosonzhang — APPROVED** [review]

**Pull request merged**